spark-benchmarks/src/main/scala/AdvertisingSpark.scala

```diff
@@ -91,20 +91,18 @@ object KafkaRedisAdvertisingStream {
     //each record in the RDD: key:(campaign_id : String, window_time: Long),  Value: (ad_id : String)
     //DStream[((String,Long),String)]
 
-    val groupedByCampaignTime = campaign_timeStamp.groupByKey()
-    // DStream[((String,Long), Iterable[String])]
+    // since we're just counting use reduceByKey
+    val totalEventsPerCampaignTime = campaign_timeStamp.mapValues(_ => 1).reduceByKey(_ + _)
```

Comment:
Oh, right to get rid of the mapValues? Ok, I see what you mean. It gets pipelined, but yes it could've been:

```campaign_timeStamp.combineByKey(_ => 1, (a: Int, b) => a+1, (a: Int, b: Int) => a+b)```

That's what you meant? Sorry, misunderstood. Thanks!